guides/keras_cv/semantic_segmentation_deeplab_v3_plus.py

```diff
@@ -0,0 +1,344 @@
+"""
+
+
+def unpackage_tfds_inputs(inputs):
+    return {
+        "images": inputs["image"],
+        "segmentation_masks": inputs["class_segmentation"],
+    }
+
+
+train_ds = train_ds.map(unpackage_tfds_inputs)
+train_ds = train_ds.map(keras_cv.layers.Resizing(height=512, width=512))
+train_ds = train_ds.batch(4, drop_remainder=True)
+
+batch = next(iter(train_ds.take(1)))
+
+keras_cv.visualization.plot_segmentation_mask_gallery(
+    batch["images"],
+    value_range=(0, 255),
+num_classes=21,  # The number of classes for the oxford iiit pet dataset. The VOC dataset
+also includes 1 class for the background.
+    y_true=batch["segmentation_masks"],
+    scale=3,
+    rows=2,
+    cols=2,
+)
+
+"""
+The preprocessing is applied to the evaluation dataset `eval_ds`. A batch of `eval_ds`
+can be visualized using `keras_cv.visualization.plot_segmentation_mask_gallery`.
+"""
+
+eval_ds = eval_ds.map(unpackage_tfds_inputs)
+eval_ds = eval_ds.map(keras_cv.layers.Resizing(height=512, width=512))
+eval_ds = eval_ds.batch(4, drop_remainder=True)
+
+batch = next(iter(eval_ds.take(1)))
+
+keras_cv.visualization.plot_segmentation_mask_gallery(
+    batch["images"],
+    value_range=(0, 255),
+num_classes=21,  # The number of classes for the oxford iiit pet dataset. The VOC dataset
+also includes 1 class for the background.
+    y_true=batch["segmentation_masks"],
+    scale=3,
+    rows=2,
+    cols=2,
+)
+
+"""
+## Data Augmentation
+
+KerasCV provides a variety of image augmentation options. In this example, we will use
+the `RandomFlip` augmentation to augment the training dataset. The `RandomFlip`
+augmentation randomly flips the images in the training dataset horizontally or
+vertically. This can help to improve the model's robustness to changes in the orientation
+of the objects in the images.
+"""
+
+train_ds = train_ds.map(keras_cv.layers.RandomFlip())
+batch = next(iter(train_ds.take(1)))
+
+keras_cv.visualization.plot_segmentation_mask_gallery(
+    batch["images"],
+    value_range=(0, 255),
+    num_classes=21,
+    y_true=batch["segmentation_masks"],
+    scale=3,
+    rows=2,
+    cols=2,
+)
+
+"""
+## Model Configuration
+
+Please feel free to modify the configurations for model training and note how the
+training results changes. This is an great exercise to get a better understanding of the
+training pipeline.
+
+The learning rate schedule is used by the optimizer to calculate the learning rate for
+each epoch. The optimizer then uses the learning rate to update the weights of the model.
+In this case, the learning rate schedule uses a cosine decay function. A cosine decay
+function starts high and then decreases over time, eventually reaching zero. The
+cardinality of the VOC dataset is 2124 with a batch size of 4. The dataset cardinality
+is important for learning rate decay because it determines how many steps the model
+will train for. The initial learning rate is 0.007 and the decay steps are 2124.
```

Comment:
It might be more compelling to just plot the learning rate over epochs.

Reply:
added